src/sync/installation.ts

```diff
@@ -412,14 +413,32 @@ export const processInstallation = (sendBackfillMessage: (message, delay, logger
 				jiraHost
 			});
 
+			let hasNextMessage = false;
+			let nextMessage: BackfillMessagePayload | undefined = undefined;
+			let nextMessageDelaySecs: number | undefined = undefined;
+			let nextMessageLogger: Logger | undefined = undefined;
+
 			const result = await deduplicator.executeWithDeduplication(
 				`i-${installationId}-${jiraHost}-ghaid-${gitHubAppId || "cloud"}`,
-				() => doProcessInstallation(data, sentry, logger, sendBackfillMessage)
+				() => doProcessInstallation(data, sentry, logger, (message, delaySecs, logger) => {
```

Comment:
So this is a current prod bug?

Reply:
But-not-bug @gxueatlassian because currently we are processing only one task and schedule the message at the very end. Till the moment the message is read by a worker, we already release the flag.

However, with this change, this is exacerbated because after the main task is over some subtasks might still be in flight.

Reply:
`Till the moment the message is read by a worker, we already release the flag.` <-- Technically, although very unlikely, that this still a problem of current prod. Like send a msg at the end of the processing, but other worker node pick that msg up, check dedupe and find out it is not released yet.

But all good, you fix it anyway 🤷 👍

Reply:
Yeah, pretty sure this happens. Not a catastrophe because it will just cause 1 minute delay. However that was quite a problem for that particular customer cause everything because sloooow.